File: OP_labs_cs/lab2_cs/lab2_cs/FileFunctions.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;

namespace lab2_cs
{
    public class FileFunctions
    {
        public static void CreateFirstFile(string fileName)
        {
            Console.WriteLine("Enter 'a' to add information to file." +
                              "\nEnter 'n' to create new file.");
            string checkFile = Console.ReadLine();
            BinaryWriter file = new BinaryWriter(File.Open(fileName, checkFile == "a" || checkFile == "A" ? FileMode.OpenOrCreate : FileMode.Create));
            file.Close();
        }
        public static void CreateSecondFile(string fileName)
        {
            BinaryWriter file = new BinaryWriter(File.Open(fileName, FileMode.Create));
            file.Close();
        }
        public static void WriteToFile(string fileName, string name, string birthday, string sex)
        {
            using (BinaryWriter writer = new BinaryWriter(File.Open(fileName, FileMode.Append)))
            {
                writer.Write(name);
                writer.Write(birthday);
                writer.Write(sex);
            }
        }
        public static List<Abit> ReadFromFile(string fileName)
        {
            List<Abit> list = new List<Abit>();
            using (BinaryReader reader = new BinaryReader(File.Open(fileName, FileMode.Open)))
            {
                while (reader.BaseStream.Position != reader.BaseStream.Length)
                {
                    string fullName = reader.ReadString();
                    int[] birthDay = ContentFunctions.SplitBirthday(reader.ReadString());
                    string Sex = reader.ReadString();
                    Abit.SexType sex = Sex == "Male" || Sex == "m" || Sex == "M" ? Abit.SexType.Male : Abit.SexType.Female;
                    Abit abit = new Abit(fullName, birthDay[0], birthDay[1], birthDay[2], sex);
                    list.Add(abit);
                }
            }

            return list;
        }
    }
}

File: OP_labs_py/lab2_py/FileFunctions.py
import pickle
from Abit import Abit


def create_file(file_name):
    check_file = input("Enter 'a' to add information to file "
                       "or enter 'n' to create new file: ")
    file = open(file_name, "ab" if check_file == "a" or check_file == "A" else 'wb')
    file.close()


def write_to_file(file_name: str, full_name: str, birthday: str, sex: str):
    with open(file_name, "ab") as file:
        pickle.dump(full_name, file)
        pickle.dump(birthday, file)
        pickle.dump(sex, file)


def read_from_file(file_name: str):
    abits_list = []
    with open(file_name, "rb") as file:
        while True:
            try:
                full_name = pickle.load(file)
                birthday = pickle.load(file)
                sex = pickle.load(file)
                abit = Abit(full_name, birthday, sex)
                abits_list.append(abit)
            except EOFError:
                break
    return abits_list


def create_recruits_file(file_name: str):
    file = open(file_name, "wb")
    file.close()
